python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -127,13 +128,17 @@ def __init__(self, provider: NodeProvider):
         self.worker_ids: List[NodeID] = []
         # The head node (node kind "head")
         self.head_id: Optional[NodeID] = None
+        # Map of replica IDs to worker nodes in each replica
+        self.replicas_to_nodes = defaultdict(list)
 
         for node in self.all_node_ids:
             node_kind = provider.node_tags(node)[TAG_RAY_NODE_KIND]
             if node_kind == NODE_KIND_WORKER:
                 self.worker_ids.append(node)
             elif node_kind == NODE_KIND_HEAD:
                 self.head_id = node
+            node_replica_index = provider.node_tags(node)[TAG_RAY_REPLICA_INDEX]
```

Comment:
Is it possible that this throws a KeyError in case `TAG_RAY_REPLICA_INDEX` is not applied? I.e. non TPU use cases

Reply:
That's a good catch, I added in a check for this.